app/src/main/java/org/akvo/flow/util/InternetDataConnection.java

```diff
@@ -0,0 +1,350 @@
+    }
+
+    /**
+     * returns the time between now and from the point that .connect() was called in milliseconds
+     * @return int delta time between now and .connect() in millis
+     */
+    public int getElapsedTime()
+    {
+        return (int) (System.currentTimeMillis() - startTime);
+    }
+
+    /**
+     * {@link HttpURLConnection#getResponseCode()}
+     */
+    public int getStatusCode() throws IOException
+    {
+        try
+        {
+            return connection.getResponseCode();
+        } catch (IOException e)
+        {
+            // HttpUrlConnection will throw an IOException if any 4XX
+            // response is sent. If we request the status again, this
+            // time the internal status will be properly set, and we'll be
+            // able to retrieve it.
+            return connection.getResponseCode();
+        }
+    }
+
+    /**
+     * checks that the statusCode is 200 (OK)
+     * throws an exception if its not
+     * @throws IOException
+     */
+    public void verifyOk() throws IOException
+    {
+        if (!isOk()) {
+            throw createHttpException();
+        }
+    }
+
+    /**
+     * return true if the statusCode is 200 (OK)
+     * @throws IOException
+     */
+    public boolean isOk() throws IOException
+    {
+        int status = getStatusCode();
+        return (status == HttpURLConnection.HTTP_OK);
+    }
+
+    /**
+     * creates a new instance of a HttpException object with the responseMessage and the statusCode
+     * @return a new HttpException object for throwing
+     * @throws IOException
+     */
+    public HttpException createHttpException() throws IOException
+    {
+        if (isOk())
+        {
+            throw new RuntimeException("Trying to create HttpException for working connection");
+        }
+        return new HttpException(getResponseMessage(), getStatusCode());
+    }
+
+    /**
+     * tells the object to disconnect the connection and close all related streams
+     * @throws IOException
+     */
+    @Override
+    public void close() throws IOException
+    {
+        if (connection != null)
+        {
+//            if (input != null)
+//            {
+//                input.get().close();
+//            }
+//            if (output != null)
+//            {
+//                output.get().close();
+//            }
+            connection.disconnect();
+        }
+    }
+
+    public HttpURLConnection getConnection()
+    {
+        return connection;
+    }
+
+    /**
+     * enables InternetDataConnection to only use input stream. <br>
+     * If you want to use both Input and Output, decorate this method with andOutput();
+     *
+     * example:
+     * {@code .connect().forInput().andOutput();}
+     *
+     * @return an object that provides you with a BufferedInputStream only
+     */
+    public InternetDataConnection.InputStreamProvider forInput() throws IOException
+    {
+        if (input != null)
```

Comment:
It is not a good practice to leave the brackets out even if there is only one line you should always have them. If someone else comes and adds another line it may produce a bug.